frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx

```diff
@@ -18,6 +20,8 @@ export const ViewHeaderContainer = styled(ViewSection)`
     flex-direction: column;
     align-items: start;
     padding: ${space(1)} 0;
+    ${({ isNavBarOpen }) =>
+      isNavBarOpen ? `margin-top: ${APP_SUBHEADER_HEIGHT};` : null}
```

Comment:
only add margin when the navbar is open because that's when we don't render the subheader.

https://github.com/metabase/metabase/blob/ddcf1fc658491e02c0daf016544ab4ac13100d20/frontend/src/metabase/nav/components/AppBar/AppBarSmall.tsx#L49